### DISCUSSION TITLE
Support for updating image tags in `from` parameter of `RUN --mount=type=bind` instructions in Dockerfile

### DISCUSSION THREAD

**Discussion by marvinruder** [renovatebot/renovate#27813] [Suggest an Idea]
### Tell us more.

The [`RUN --mount=type=bind`](https://docs.docker.com/reference/dockerfile/#run---mounttypebind) syntax allows mounting files or directories of images into a Docker build step. One advantage over the already supported `COPY --from=<tag>` syntax is that the number of layers can be reduced when copying multiple files from the same image:

```Dockerfile
# Produces 3 layers
COPY --from=node:21.7.0-alpine /etc/group /etc
COPY --from=node:21.7.0-alpine /etc/passwd /etc
COPY --from=node:21.7.0-alpine /usr/local/bin/node /usr/local/bin

# vs.

# Produces 1 layer
RUN \
  --mount=type=bind,from=node:21.7.0-alpine,source=/,target=/mnt \
  cp -a /mnt/etc/group /etc/group && \
  cp -a /mnt/etc/passwd /etc/passwd && \
  cp -a /mnt/usr/local/bin/node /usr/local/bin/node
```

Recently, I discovered that Renovate updates Docker image tags in `COPY --from=<tag>` instructions, but not in `RUN --mount=type=bind,from=<tag>` instructions.

I would like Renovate to support both cases.

**Comment by rarkins** [comment]
Would be easier to support if we limit it to having the RUN and image on a single line, e.g. `RUN --mount=type=bind,from=node:21.7.0-alpine` without a line break

PR welcome if you wish to contribute this

**Reply by viceice** [reply]
@rarkins our parser already handles multiline, we just need to handle the run argument
